### PR TITLE
define a new Go type Port for consolidation 

### PR DESCRIPTION
**Pull request by hbagdi**
- Defines a new type Port
- Re-use the type across all instances of port
- Change the type from int32 to uint32

### Note

I spotted this while working on #422.
Please review only the last commit here. To avoid conflicts, this PR builds on top of #422 and should be merged after that lands in.

### CODE REVIEW

File: apis/v1alpha1/route_types.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ type RouteGateways struct {
 	GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
 }
 
+// Port defines a network port.
+//
+// +kubebuilder:validation:Minimum=1
+// +kubebuilder:validation:Maximum=65535
+type Port uint32
```
Comment (jpeach) [inline apis/v1alpha1/route_types.go:62]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

> Do not use unsigned integers, due to inconsistent support across languages and libraries. Just validate that the integer is non-negative if that's the case.

The validation keeps it in range, so it should be OK to leave this as `int32`.

Comment (jpeach) [inline apis/v1alpha1/route_types.go:62]: Suggest that we call this type `PortNumber` for clarity.

Comment (hbagdi) [reply apis/v1alpha1/route_types.go]: `PortNumber` feels a little more verbose than it needs to be. What is the reasoning behind the suggestion?

Comment (robscott) [reply apis/v1alpha1/route_types.go]: In k/k our [Port types include a ton of fields](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L3702), including a name and protocol. I think using `PortNumber` here may make the simplicity of this type more clear.

Comment (hbagdi) [reply apis/v1alpha1/route_types.go]: This is a type alias and not even a struct. Do we still need to be explicit even then?

Comment (robscott) [reply apis/v1alpha1/route_types.go]: It's a small detail, I'd just imagine that anyone browsing through the types might see `Port` in a `forwardTo` and wonder if that included all the things that `ServicePort` did or if it was just a port number. Obviously it's not difficult to figure that out, just takes an extra step.

Comment (hbagdi) [reply apis/v1alpha1/route_types.go]: I see. I've updated this to be `PortNumber`. Like you said, it is a small detail and I don't feel too strongly either ways.

**Comment by robscott** [comment]
Thanks!

/lgtm
/approve

**Comment by robscott** [comment]
Sorry, just had to double check that a port was intentionally optional.

/hold cancel

File: apis/v1alpha1/route_types.go (hunk 1 above)
Comment (jpeach) [reply apis/v1alpha1/route_types.go]: Yeh, it's clarity ... `PortNumber` is immediately obvious and evocative of its meaning.

**Pull request merged**